pkg/lightning/backend/tidb/tidb.go

```diff
@@ -364,31 +378,99 @@ func (be *tidbBackend) ImportEngine(context.Context, uuid.UUID) error {
 }
 
 func (be *tidbBackend) WriteRows(ctx context.Context, _ uuid.UUID, tableName string, columnNames []string, rows kv.Rows) error {
-	var err error
-outside:
+rowLoop:
 	for _, r := range rows.SplitIntoChunks(be.MaxChunkSize()) {
+		batch := true
+		rowsToSkip := make(map[int]struct{})
+	retryLoop:
 		for i := 0; i < writeRowsMaxRetryTimes; i++ {
-			err = be.WriteRowsToDB(ctx, tableName, columnNames, r)
+			errRowIdx, err := be.WriteRowsToDB(ctx, tableName, columnNames, r, batch, rowsToSkip)
```

Comment:
Please check the `TestWriteRowsErrorSkip` test case in this PR, I think it can help understand the current error count and retry strategy. Maybe it's not quite like what you think here.